utils/api/apiEndpointWrapper.ts

```diff
@@ -0,0 +1,90 @@
+import { NextResponse } from "next/server";
```

Comment:
The idea here is to create a higher order function to wrap all of the API endpoints for a couple of things:

Consistency in response schema for status codes, success, errors etc. Makes it easier on the UI to have a structured format to expect the response in and for displaying errors we have more control on how errors are returned by enforcing this with TS. Can easily map errors to form fields they belong to.
Decoupling authorization logic from business logic. Makes it easier to share common validation logic across functions and there is a clear separation of concern from the authorization logic and the business logic. Similar to OPA which I have used before in the context of spring boot web services
Reduce repeat boilerplate code like loading in prismas, wrapping in try catch for error handling etc.
There are some cons to something like this since it is custom it is now one extra complexity that would need to be maintained on our end and it might be a little harder for newer devs to wrap their head around.

For this assignment though, this abstraction made sense to enforce API endpoint consistency